stdlib/public/stubs/Stubs.cpp

```diff
@@ -487,6 +487,12 @@ static const char *_swift_stdlib_strtoX_clocale_impl(
     const char * nptr, T* outResult, T huge,
     T (*posixImpl)(const char *, char **, locale_t)
 ) {
+  if (strcasecmp(nptr, "snan") == 0 || strcasecmp(nptr, "+snan") == 0 ||
```

Comment:
K, and this logic will need to either be copied to precede the `istringstream` or you'll need to extract it.